File: trunk/PO-8_210639/task_07/src/goodfacebuilder.cpp
#include "goodfacebuilder.h"
#include <qDebug>

void goodFaceBuilder::buildEars() {
	ears* newEars = new ears();
	newEars->color = "beidge";
	newEars->figure = "circle";
	Ears = newEars;
}

void goodFaceBuilder::getResult() const
{
	qDebug() << "good face: ";
	Eyes->print();
	Ears->print();
	Mouth->print();
	Nose->print();
	Hair->print();
}

void goodFaceBuilder::buildEyes() {
	eyes* newEyes = new eyes();
	newEyes->color = "blue";
	newEyes->figure = "circle";
	Eyes = newEyes;
}

void goodFaceBuilder::buildMouth() {
	mouth* newMouth = new mouth();
	newMouth->color = "pink";
	newMouth->figure = "half circle";
	Mouth = newMouth;
}

void goodFaceBuilder::buildNose() {
	nose* newNose = new nose();
	newNose->color = "beidge";
	newNose->figure = "triangle";
	Nose = newNose;
}

void goodFaceBuilder::buildHair() {
	hair* newHair =  new hair();
	newHair->color = "blonde";
	newHair->figure = "rectangle";
	Hair = newHair;
}

File: trunk/PO-8_210648/task_07/ConsoleApp1/ConsoleApp2/GoodFaceBuilder.cs
namespace ConsoleApp2;

public class GoodFaceBuilder : Builder
{
    private Product _product = new Product();

    public override void BuildEyes()
    {
        _product.AddEyes(new string[]{"good","blue"});
    }

    public override void BuildNose()
    {
        _product.AddNose(new string[]{"good","blue"});
    }

    public override void BuildMouth()
    {
        _product.AddMouth(new string[]{"good","blue"});
    }

    public override void BuildEars()
    {
        _product.AddEars(new string[]{"good","blue"});
    }

    public override void BuildHair()
    {
        _product.AddHair(new string[]{"good","blue"});
    }

    public override Product GetResult()
    {
        return _product;
    }
}